src/lib/commandclass/ColorSwitchCC.ts

```diff
@@ -0,0 +1,446 @@
+export class ColorSwitchCCSupportedReport extends ColorSwitchCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+
+		// Docs say 'variable length', but the table shows 2 bytes.
+		validatePayload(this.payload.length >= 2);
+
+		// Don't know what order these come in.
+		//  Blindly assuming its in order of payload,
+		//  but the table seems to be big-endian.
+		this._supportsWarmWhite = Boolean(this.payload[0] & 1);
+		this._supportsColdWhite = Boolean(this.payload[0] & 2);
+		this._supportsRed = Boolean(this.payload[0] & 4);
+		this._supportsGreen = Boolean(this.payload[0] & 8);
+		this._supportsBlue = Boolean(this.payload[0] & 16);
+		this._supportsAmber = Boolean(this.payload[0] & 32);
+		this._supportsCyan = Boolean(this.payload[0] & 64);
+		this._supportsPurple = Boolean(this.payload[0] & 128);
+		this._supportsIndex = Boolean(this.payload[2] & 1);
+
+		//this.persistValues();
+	}
+
+	private _supportsWarmWhite: boolean;
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyBoolean,
+		label: "Supports Warm White",
+	})
+	public get supportsWarmWhite(): boolean {
+		return this._supportsWarmWhite;
+	}
+
+	private _supportsColdWhite: boolean;
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyBoolean,
+		label: "Supports Cold White",
+	})
+	public get supportsColdWhite(): boolean {
+		return this._supportsColdWhite;
+	}
+
+	private _supportsRed: boolean;
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyBoolean,
+		label: "Supports Red",
+	})
+	public get supportsRed(): boolean {
+		return this._supportsRed;
+	}
+
+	private _supportsGreen: boolean;
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyBoolean,
+		label: "Supports Green",
+	})
+	public get supportsGreen(): boolean {
+		return this._supportsGreen;
+	}
+
+	private _supportsBlue: boolean;
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyBoolean,
+		label: "Supports Blue",
+	})
+	public get supportsBlue(): boolean {
+		return this._supportsBlue;
+	}
+
+	private _supportsAmber: boolean;
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyBoolean,
+		label: "Supports Amber",
+	})
+	public get supportsAmber(): boolean {
+		return this._supportsAmber;
+	}
+
+	private _supportsCyan: boolean;
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyBoolean,
+		label: "Supports Cyan",
+	})
+	public get supportsCyan(): boolean {
+		return this._supportsCyan;
+	}
+
+	private _supportsPurple: boolean;
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyBoolean,
+		label: "Supports Purple",
+	})
+	public get supportsPurple(): boolean {
+		return this._supportsPurple;
+	}
+
+	private _supportsIndex: boolean;
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyBoolean,
+		label: "Supports Indexed Color",
+	})
+	public get supportsIndex(): boolean {
+		return this._supportsIndex;
+	}
+
+	public toJSON(): JSONObject {
```

Comment:
I've started to deprecate the toJSON method in favor of the `toLogEntry` method. You can find an example in MultiChannelCC, lines 840ff